pkg/network/server.go

```diff
@@ -1112,22 +1111,26 @@ func (s *Server) handleGetAddrCmd(p Peer) error {
 // 2. Send requests for chunk in increasing order.
 // 3. After all requests were sent, request random height.
 func (s *Server) requestBlocks(bq blockchainer.Blockqueuer, p Peer) error {
-	var currHeight = bq.BlockHeight()
+	pl := s.getRequestBlocksPayload(p, bq.BlockHeight(), &s.lastRequestedHeight)
+	return p.EnqueueP2PMessage(NewMessage(CMDGetBlockByIndex, pl))
+}
+
+func (s *Server) getRequestBlocksPayload(p Peer, currHeight uint32, lastRequestedHeight *atomic.Uint32) *payload.GetBlockByIndex {
```

Comment:
If it no longer uses `(s *Server)` then it doesn't need it (can be a function instead of a method).